security/fma-shared-lockbox.md

```diff
@@ -0,0 +1,136 @@
+    | Case 7 | no | yes | yes | Same as Case 3: The transaction will revert because the `OptimismPortal` was not allow-listed since `addDependency` was not executed first. | Re-execute the complete batch process. |
+    
+    It’s worth noting that the process is designed to revert if the step 2 (`upgradeAndCall`) is executed before the step 1 (`addDependency`). Consequently, only a few additional cases can be described when the order is also altered.
+    
+    | Case N | Type | Expected result | Recovery Path |
+    | --- | --- | --- | --- |
+    | Case 8 | Step 1 → Step 3 → Step 2 | Same as Case 5, the `OptimismPortal` proxy implementation will remain incorrect (the latest applied implementation would be the `LiquidityMigrator`). As a result, none of the expected functions of the `OptimismPortal` will work. | Re-execute the step 3 only. |
+    | Case 9 | Step 3 → Step 1 → Step 2 | ETH will be migrated to the `SharedLockbox`, but the `OptimismPortal` proxy implementation will remain incorrect (the latest applied implementation would be the `LiquidityMigrator`). As a result, none of the expected functions of the `OptimismPortal` will work. | Re-execute the step 3 only. |
+    | Case 10 | Step 3 → Step 1 (Step 2 missed) | Same as Case 6, `OptimismPortal` will be able to lock and unlock ETH from the `SharedLockbox`, but it will still store its ETH in the `OptimismPortal`. ETH withdrawals are at risk to fail for any `OptimismPortal` in the cluster because a portion of the expected ETH was not migrated and remains in the latest added chain. | Execute a batch TX with the steps 2 and 3. |
+
+### Generic items we need to take into account:
+
+See [fma-generic-contracts.md](https://github.com/ethereum-optimism/design-docs/blob/main/security/fma-generic-contracts.md).
+
+- [x]  Check this box to confirm that these items have been considered and updated if necessary.
+
+## Action Items
+
+- Resolve all the comments.
+- Proceed with the code implementation and make any necessary approved changes.
+- Move into testing of all the components.
+- Make sure there are monitoring solutions ready to cover all the cases.
+
+## Audit Requirements
+
+The `OptimismPortalInterop.sol`, `SharedLockbox.sol`, `LiquidityMigrator.sol`, `SystemConfigInterop.sol`, and `SuperchainConfig.sol` require an audit before going to production. 
+
+## Additional Notes
+
+- Most high-risk cases involving the `SharedLockbox` pertain to the unexpected extraction of ETH liquidity. There are a few possible approaches to minimize the risks of such events:
+    - Adding up a timelock to the `finalizeWithdrawalTransaction` function, which would delay fund extraction.
+    - Introducing a cooldown period for each newly authorized `OptimismPortal`. During the cooldown, the new portal would be unable to successfully call `unlockETH` and `lockETH`.
+    
+    In both cases, these time periods would allow the Guardian to pause the system before a malicious attempt is successful.
```

Comment:
Why are these mitigations only briefly discussed in "Additional Notes"? Have we considered them in the design doc process and decided not to do them for some reason? Or is the FMA revealing we want to add this into scope? If the former, this should be the Intro section (either directly or via link) so the reviewer knows why we don't have these in the design. If the latter this mitigations should be tracked as action items. And in either case I don't think this section is the right place for this info as a result :)